### PR TITLE
[APEX-230] Set line length to 120

### PR DESCRIPTION
**Pull request by ilganeli**
- Set line length to 120 in CheckStyle. 
- Added enforcement of annotations being on separate lines.
- Increased wrapping indentation to 4.
- I did not update all relevant CodeStyle settings to automatically enforce wrapping for longer lines because how we want to do this is a more complex question and potentially outside the scope of this issue. For now, CheckStyle will simply throw an error if the line is too long. In IntelliJ, two settings were necessary to have things work correctly:
  1) In Chained Method Calls I set "wrapIfLong"
  2) Ensure right margin is not exceeded checked on.

### CODE REVIEW

**Comment by vrozov** [comment]
@ilganeli Please squash commits related to CheckStyle into a single commit and add reference to JIRA for every commit. It may still be a single pull request, but in general it will be better to have 2 different PR, so discussion/review of code style changes in StreamingContainer.java or any other file does not mix with discussion on CheckStyle rules.

For checkstyle rules I think we need to agree on increasing wrapping ident to 4 and enforcing annotation placement. It will be also good to combine this with upgrade to 6.12 checkstyle.

**Comment by otterc** [comment]
@ilganeli NoWhitespaceBefore Comma issue has been fixed in 6.11. What error was it throwing?

File: engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
  *
- *   http://www.apache.org/licenses/LICENSE-2.0
```
Comment (tweise) [inline engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java:10]: Cannot modify license headers. License headers need to be consistent across all file types and comply with automated license check. This is not just an IDE concern.

**Comment by tweise** [comment]
Ilya, the intention behind https://malhar.atlassian.net/browse/APEX-207 is to address code style in a comprehensive manner. Let's not work on individual files and individual rules. I believe @chandnisingh is planning to take it up. Personally, I don't think we can expect to reformat entire files in one IDE and match exactly what another IDE would produce. We should instead agree on priority basis the basic set of rules that we want to enforce via checkstyle and then fix existing violations.

**Comment by ilganeli** [comment]
@tweise The only reason I submitted the additional changes is that otherwise the number of CheckStyle violation exceeded the allowed limit causing it to fail in travis. 

@chandnisingh The error is below:
`The Checkstyle rules file could not be loaded.
           cannot initialize module TreeWalker - Token "COMMA" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceBeforeCheck
           The file has been blacklisted for 60s. (show balloon)`

**Comment by tweise** [comment]
Understood. We need to make further checkstyle changes in conjunction with APEX-207 then.

**Comment by ilganeli** [comment]
@tweise What would you recommend as the best path forward for this patch? Remove the additional changes and leave the changes only limited to the checkstyle updates?

**Comment by otterc** [comment]
@ilganeli @tweise 
I intentionally introduced a space and ran checkstyle and I don't get the error which Ilya has posted. This check is necessary. 

[ERROR] src/main/java/com/datatorrent/stram/engine/StreamingContainer.java:[302,68](whitespace) NoWhitespaceBefore: ',' is preceded with whitespace.

**Comment by otterc** [comment]
If the Checkstyle rules was unable to load then we would not have been able to build APEX at all. This maybe a problem just with travis

**Comment by vrozov** [comment]
@ilganeli Ilya, I suggest to drop commits related to StreamingContainer.java from this PR and add few more changes to checkstyle:
1. Upgrade to 6.12. Verify that  NoWhitespaceBefore allows COMMA in that version.
2. Enforce annotations on a separate line.
3. Increase wrapping indentation to 4.

This should cover all known issues with checkstyle plugin, so we can move forward with fixing code style violations in the existing code. The later should be done in one or few batch updates as voted by the community.

**Comment by ilganeli** [comment]
@vrozov @tweise I've made the suggested updates and updated the comment for this PR accordingly.

**Comment by vrozov** [comment]
@ilganeli Please update number of allowed code violations, as otherwise build fails. In the commit comments please include APEX-248 and APEX-249.

Did you try upgrading checkstyle to 6.12?

**Comment by ilganeli** [comment]
@vrozov Where do I configure the allowed number of violations? I did update checkstyle and confirmed that it works with the latest version.

**Comment by vrozov** [comment]
Number of violations are configured in modules pom.xml and project pom has configuration for the checkstyle version to use:

```
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-checkstyle-plugin</artifactId>
        <configuration>
          <maxAllowedViolations>2072</maxAllowedViolations>
        </configuration>
      </plugin>
```

and

```
        <plugin>
          <groupId>org.apache.maven.plugins</groupId>
          <artifactId>maven-checkstyle-plugin</artifactId>
          <version>2.17</version>
          <dependencies>
            <dependency>
              <groupId>com.puppycrawl.tools</groupId>
              <artifactId>checkstyle</artifactId>
              <version>6.11.2</version>
            </dependency>
          </dependencies>
```

**Comment by vrozov** [comment]
After upgrade to 6.12 we can also enforce anonymous classes code style. Can you please verify that this rule is either on by default or turn it on.

**Comment by otterc** [comment]
Here is the ticket for Anonymous class brace placement
https://malhar.atlassian.net/browse/APEX-227

and the ticket for upgrading checkstyle version
https://malhar.atlassian.net/browse/APEX-239

File: .idea/codeStyleSettings.xml [hunk 2]
```diff
@@ -102,5 +104,4 @@
     </option>
     <option name="USE_PER_PROJECT_SETTINGS" value="true" />
   </component>
-</project>
-
+</project>
```
Comment (243826) [inline .idea/codeStyleSettings.xml]: end the files with a new line character.

File: apex_checks.xml [hunk 3]
```diff
@@ -97,6 +97,10 @@
       <message key="ws.notPreceded" value="WhitespaceAround: ''{0}'' is not preceded with whitespace."/>
     </module>
 
+    <module name="LineLength">
+      <property name="max" value="120"/>
```
Comment (243826) [inline apex_checks.xml:101]: what's the source of this magic number? why not 80, or 100 or 150? There is some good discussion here: http://stackoverflow.com/questions/578059/studies-on-optimal-code-width

Comment (ilganeli) [inline apex_checks.xml:101]: This was agreed in discussion on the mailing list.

Thank you,
Ilya Ganelin

-----Original Message-----

Comment (243826) [inline apex_checks.xml:101]: Thanks and sorry. I see that discussion now.

File: .idea/codeStyleSettings.xml (hunk 2 above)
Comment (vrozov) [inline .idea/codeStyleSettings.xml]: The file is managed by IntelliJ and it does not end the file with EOL. I suggest we keep as is.

**Comment by ilganeli** [comment]
@chandnisingh Fixed all issues except for APEX-239

**Comment by otterc** [comment]
@ilganeli see following issues:
The pom of all child modules with existing failures need to be corrected and is causing failure of the pull request build
1. api has 103 existing errors than 102 
2. common has 124 existing errors. Currently 84
3. bufferserver has 165 existing errors. Currently 83.
4. engine has 2942 existing errors. currently 2072.

I think another change that should accompany this is:
 .idea/codeStyleSettings.xml should have  the continuation indentation as '4'

**Comment by ilganeli** [comment]
Sigh - I don't understand what I've done, it looks like something got messed up in the last change when I merged in the latest devel-3. Any thoughts on how to fix the problem?

**Pull request merged**